src/corpus/utils/repositories.rs

```diff
@@ -0,0 +1,167 @@
+//! # Utility Functions for Git Repositories
+
+use std::{collections::HashMap, sync::Mutex};
+
+use lazy_static::lazy_static;
+use reqwest::blocking::Client;
+use serde::Deserialize;
+
+use crate::corpus::errors::DownloaderError;
+
+lazy_static! {
```

Comment:
I know this is pretty ugly, but I can't think of a better / cleaner solution right now. The other option would be to pass a `HashMap` through every single function down until `download_tarball` in `downloader.rs`, and I feel like that's worse than having a global variable.